### PR TITLE
ASRC-280: split out CalculationResult

### PR DESCRIPTION
**Pull request by dgtombs**
from the Calculation class.

Also support only 1 ProcedureVariable per calculation. We had partial support for it in the code before.

### CODE REVIEW

File: srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/Calculation.java [hunk 1]
```diff
@@ -96,7 +102,40 @@ public void setSpecialty(final Specialty specialty)
...
         }
         
-        return fSpecialty.getModelVariables();
+        final ImmutableSet<Variable> vars = fSpecialty.getModelVariables();
+        // Check our method contract.
+        if (hasMultipleProcedureVariables(vars))
+        {
+            // See method contract.
+            throw new ConfigurationException(
+                    "The specialty requires multiple ProcedureVariables");
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/Calculation.java]: Shouldn't this read more like, "Specialties cannot have more than one ProcedureVariables" to make it clear what the error is trying to say.

File: srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            final DateTime startDateTime,
+            final int patientDfn,
+            final String specialtyName,
+            final Set<Value> values,
+            final Map<String, Double> outcomes)
+    {
+        fStartDateTime = startDateTime;
+        fPatientDfn = patientDfn;
+        fSpecialtyName = Objects.requireNonNull(
+                specialtyName, "specialty name must be non-null");
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java]: We have a limit in the database creation that specifies specialties are 255 characters, but what about in Java code?

File: srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            }
+            else
+            {
+                otherValuesBuilder.add(v);
+            }
+        }
+        fProcedureValue = procedureValue;
+        fNonProcedureValues = otherValuesBuilder.build();
+
+        fOutcomes = ImmutableSortedMap.copyOf(
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java]: I like the early separation of values.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/ValueDisplayOrder.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class ValueDisplayOrder implements Comparator<Value>
+{
+    @Override
+    public int compare(final Value a, final Value b)
+    {
+        // Sort ProcedureValues first.
+        final int aIsProcedure = (a instanceof ProcedureValue) ? 1 : 0;
+        final int bIsProcedure = (b instanceof ProcedureValue) ? 1 : 0;
+        // Subtract b from a to because a ProcedureValue is _less than_ a non-
```
Comment (jamesbigmac) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/ValueDisplayOrder.java]: What's with the underscores?

File: srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java (hunk 2 above)
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java]: Done.

**Pull request merged**